easybuild/easyblocks/s/scalapack.py

```diff
@@ -190,8 +190,21 @@ def build_step(self):
             extra_makeopts.append('CDEFS="-D%s"' % interface)
 
         # update make opts, and build_step
+        saved_buildopts = self.cfg['buildopts']
+
+        # Only build the library first, that can be done in parallel
+        # But the library itself will be broken
+        self.cfg.update('buildopts', 'lib')
+        self.cfg.update('buildopts', ' '.join(extra_makeopts))
+
+        super(EB_ScaLAPACK, self).build_step()
```

Comment:
@akesandgren Every now and then, a `ScaLAPACK` build fails here with `ranlib: ../../../libscalapack.a: Malformed archive`, see test report in https://github.com/easybuilders/easybuild-easyconfigs/pull/5331.

To remedy this, we should run the parallel `make` such that we just ignore the exit code, rather than just calling out to `build_step` of the parent here?

Reply:
Ahh, I was looking at the failed reports but couldn't see any reason for the fail.
Ignoring the exit code might be bad in case some compile actually fails (in the future).
But one could patch the makefiles to have the ar/ranlib be ignored. And if so i should probably be done in the easyblock.

Thoughts on that?

Reply:
Well, even if the parallel `make` fails for a different reason, that's OK, since we'll run a `make -j 1` anyway (in which case it'll fail again)?

Reply:
Yes, that's true. So we could indeed ignore the first parallel make. How?

Reply:
rather than doing the `super` call, you'll need to run `make` yourself, as follows:

```python
# deliberately ignore exit code using log_ok=False
# creating libscalapack.a may fail in parallel, but should work fine with non-parallel make afterwards
run_cmd("make -j %s" % self.cfg['parallel'], log_ok=True)
```

Note: this ignores `prebuidopts` & `buildopts`, if you want to be able to control the `make` command via the easyconfig, you should make this a bit smarter, see the `ConfigureMake.build_step` implementation.

Strictly speaking, not doing so would be a regression...

Reply:
:-( I was hoping there would be a ignore_fails flag to that build step ...
Oh well, I'll see what i can come up with...

Reply:
Well, you can enhance the existing `ConfigureMake.build_step` implementation to allow ignoring failures of the `make` command it runs, optionally of course.

It's a pretty specific use case though, you usually really want to check the exit code. :)